desc/objectives/_coils.py

```diff
@@ -128,7 +128,10 @@ def _prune_coilset_tree(coilset):
 
         # map grid to list of length coils
         if grid is None:
-            grid = [LinearGrid(N=2 * c.N + 5, endpoint=False) for c in coils]
+            grid = []
+            for c in coils:
+                NFP = c.NFP if hasattr(c, "NFP") else 1
+                grid.append(LinearGrid(N=2 * c.N + 5, NFP=NFP, endpoint=False))
```

Comment:
@dpanici I don't think we want this change. Only `CoilSet` has an `NFP` attribute, and `coils` here is always a list of `Coil` objects, so this will never get called. And it would be bad if it did, because NFP>1 would mean that the grid wouldn't span the entire length of the coils (remember we use the zeta coordinates of the grid to represent the curve parameter s).

Reply:
`FourierRZCoil` also has `NFP`

Reply:
I have not thought about that though, for a `FourierRZCurve` biot savart law... yea I will revert this as it probably has unintended consequences for that calculation

Reply:
OK I forgot about `FourierRZCoil`. But I still think for these objectives that inherit from `_CoilObjective` like `CoilLength` etc. we always want the grid to span the entire length of the coil.

Reply:
Food for thought: should this actually be
```python
 for c in coils:
      NFP = c.NFP if hasattr(c, "NFP") else 1
      grid.append(LinearGrid(N=2 * c.N * NFP + 5, NFP=1, endpoint=False))
```

Since say a `FourierRZCoil` has `N=5` and `NFP=19`... without this change the coil would have only `LinearGrid(N=15)` which is `31` points for an effective toroidal resolution of `NFP*5 = 95`

Reply:
@ddudt

Reply:
I think it should be

Reply:
I think you are right, good catch. But then we also need to make this same change [here](https://github.com/PlasmaControl/DESC/blob/master/desc/geometry/core.py#L116) (and maybe other places).

Reply:
maybe just in the .compute_magnetic_field method of `FourierRZCoil` would be enough, then we basically cover everywhere that would require this (since none of the data index compute quantities care about having `NFP>1` on the grid,they work fine)

Reply:
My main point was that since you already changed it in `_CoilObjective.build` we also had to change it in `Curve.compute` for consistency, since `_CoilObjective` basically just calls `Curve.compute`.